src/exchange/order.rs

```diff
@@ -96,6 +99,34 @@ impl ClientOrderRequest {
             hashed_order_type.0,
             hashed_order_type.1,
         ))
+        
+    }
+
+    pub(crate) fn create_hashable_tuple_with_cloid(
+        &self,
+        coin_to_asset: &HashMap<String, u32>,
+    ) -> Result<(u32, bool, u64, u64, bool, u8, u64, [u8;16])> {
+        let hashed_order_type = self.order_type.get_type()?;
+        let &asset = coin_to_asset.get(&self.asset).ok_or(Error::AssetNotFound)?;
+        
+        // cloid is Some("0x1234567890abcdef1234567890abcdef".to_string()) is a 128 hex string
+        match &self.cloid {
```

Comment:
nit: prefer `if let Some(cloid) = &self.cloid { ... } else { ... }` for optional values